src/game_inventory.cpp

```diff
@@ -401,6 +401,24 @@ item_location game::inv_map_splice( const item_filter &filter, const std::string
                          title, radius, none_message );
 }
 
+class liquid_inventory_filter_preset : public inventory_filter_preset
+{
+    public:
+        explicit liquid_inventory_filter_preset( const item_location_filter &filter ) :
```

Comment:
The `filter` parameter can be moved to a `is_shown` override instead (and the class can then inherit from inventory_selector_preset` instead of `inventory_filter_preset`.

Reply:
honestly not sure how i would do that, but i ll take a look

Reply:
`inventory_filter_preset` just calls `filter` in its `is_shown` method, so you can just override `is_shown` directly and avoid the inheritance from `inventory_filter_preset`.

Reply:
i get what you want me to do (should be pretty much this):
`class liquid_inventory_filter_preset : public inventory_selector_preset
{
    public:
        explicit liquid_inventory_filter_preset( const item_location_filter &filter ) :
            inventory_selector_preset() {
            append_cell( []( const item_location & loc ) {
                if( loc.get_item() ) {
                    return string_format( "%s %s", format_volume( loc.get_item()->max_containable_volume() ),
                                          volume_units_abbr() );
                }
                return std::string( "" );
            }, _( "Storage (L)" ) );
        }

        bool is_shown( const item_location &location ) const override {
            return filter( location );
        }

    private:
        item_location_filter filter;
};`

my problem is that it throws an exception in
throw.cpp :: RaiseException(EH_EXCEPTION_NUMBER, EXCEPTION_NONCONTINUABLE, _countof(parameters), parameters);

Reply:
Actually I mean something like this:
```c++
class liquid_inventory_filter_preset : public inventory_selector_preset
{
    private:
        const item &liquid;
        const item *const avoid;

    public:
        explicit liquid_inventory_filter_preset( const item &liquid, const item *const avoid )
            : liquid( liquid ), avoid( avoid ) {
            append_cell( ... );
        }

        bool is_shown( const item_location &location ) const override {
            if( location.get_item() == avoid ) {
                return false;
            }
            if( location.where() == item_location::type::character ) {
                Character *character = g->critter_at<Character>( location.position() );
                if( character == nullptr ) {
                    debugmsg( "Invalid location supplied to the liquid filter: no character found." );
                    return false;
                }
                return location->get_remaining_capacity_for_liquid( liquid, *character ) > 0;
            }
            const bool allow_buckets = location.where() == item_location::type::map;
            return location->get_remaining_capacity_for_liquid( liquid, allow_buckets ) > 0;
        }
};
```
This way we don't have to pass a lambda around.

Reply:
could you come to the discord, so we can talk about this a bit easier?

Reply:
I don't have a discord account, sorry. Anyway, the current code already works and this is only my suggestion, so feel free to accept or reject it.

Reply:
ok, my problem with your solution is that it pretty much ends up duplicating code and adds multiple locations for a debugmsg (although that could easily be changed)

Reply:
It doesn't duplicate code, you can remove the originall lambda after my suggested change.

Reply:
sorry for the delay (got a bit sidetracked) ... you are right, i don't know why i thought that the lambda was used for more than  just the call.

I changed the code to now extend inventory_selector_preset